### PR TITLE
Add dynamic columns to tracks

### PR DESCRIPTION
**Pull request by qgp**
@jgrosseo @shahor02 @ktf Could you have a look whether this make sense to you? Should we add something else?

In contrast to AliRoot, there is no protection against NaN results and, in fact, returning NaN when the operations can't be carried out is the most consistent behaviour, I think.

Do we have any place to document the data model and what various columns are supposed to be? For some it might be obvious, for others the precise meaning might be less clear.

### CODE REVIEW

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 1]
```diff
@@ -61,6 +61,30 @@ DECLARE_SOA_DYNAMIC_COLUMN(Phi, phi, [](float snp, float alpha) -> float { retur
 DECLARE_SOA_DYNAMIC_COLUMN(Eta, eta, [](float tgl) -> float { return log(tan(0.25 * M_PI - 0.5 * atan(tgl))); });
 DECLARE_SOA_DYNAMIC_COLUMN(Pt, pt, [](float signed1Pt) -> float { return fabs(1.0 / signed1Pt); });
 DECLARE_SOA_DYNAMIC_COLUMN(Charge, charge, [](float signed1Pt) -> short { return (signed1Pt > 0) ? 1 : -1; });
+DECLARE_SOA_DYNAMIC_COLUMN(Px, px, [](float signed1Pt, float snp, float alpha) -> float {
+  double pt = 1. / std::abs(signed1Pt);
+  double cs = std::cos(alpha);
+  double sn = std::sin(alpha);
+  double r = std::sqrt((1. - snp) * (1. + snp));
+  return pt * (r * cs - snp * sn);
+});
```
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: Can these columns already depend on each other? In this case I would suggest to use directly pt as input to this one, not signed1Pt

Comment (ktf) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: for the moment it's not possible.

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 2]
```diff
@@ -61,6 +61,30 @@ DECLARE_SOA_DYNAMIC_COLUMN(Phi, phi, [](float snp, float alpha) -> float { retur
...
+DECLARE_SOA_DYNAMIC_COLUMN(Py, py, [](float signed1Pt, float snp, float alpha) -> float {
+  double pt = 1. / std::abs(signed1Pt);
+  double cs = std::cos(alpha);
```
[the remark this answers is not in the document]
Comment (qgp) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: changed to `o2::utils::sincosf(...)`

File: Framework/Core/include/Framework/AnalysisDataModel.h [hunk 3]
```diff
@@ -63,15 +64,15 @@ DECLARE_SOA_DYNAMIC_COLUMN(Pt, pt, [](float signed1Pt) -> float { return fabs(1.
...
+  utils::sincosf(alpha, sn, cs);
   double r = std::sqrt((1. - snp) * (1. + snp));
   return pt * (r * cs - snp * sn);
 });
 DECLARE_SOA_DYNAMIC_COLUMN(Py, py, [](float signed1Pt, float snp, float alpha) -> float {
   double pt = 1. / std::abs(signed1Pt);
-  double cs = std::cos(alpha);
-  double sn = std::sin(alpha);
+  float cs, sn;
+  utils::sincosf(alpha, sn, cs);
```
Comment (ktf) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: Why do we have our own wrapper to `::sincosf`?

Comment (ktf) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: Ah, ok... GNU_SOURCE

Comment (shahor02) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: yes, it is not portable...

Comment (qgp) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: Might be related to some GPU off-loading, maybe @davidrohr or @shahor02 can comment?

Comment (ktf) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: It does not work on Mac too...

Comment (qgp) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: Yes, also tried that, so I think we leave it to the os::utils version, no?

**Comment by ktf** [comment]
Notice that no test will be done until you remove the Draft state.

**Pull request merged**